### PR TITLE
converter: allow controlling referrer with option

### PR DESCRIPTION
**Pull request by imeoer**
```
Some registry implementations like Docker Hub don't support the
OCI reference type (aka referrer) yet, we shouldn't enable it by
default, instead, provide a `WithReferrer` option to control it.
```

To fix https://github.com/dragonflyoss/image-service/issues/1363

### CODE REVIEW

File: pkg/converter/types.go [hunk 1]
```diff
@@ -107,6 +107,10 @@ type MergeOption struct {
 	OCI bool
 	// OCIRef enables converting OCI tar(.gz) blob to nydus referenced blob.
 	OCIRef bool
+	// WithReferrer associates a reference to the original OCI manifest.
+	// See the `subject` field description in
+	// https://github.com/opencontainers/image-spec/blob/main/manifest.md#image-manifest-property-descriptions
+	WithReferrer bool
```
Comment (changweige) [inline pkg/converter/types.go:113]: Could help introduce why nydus image has to refer to the original OCI image?

Comment (imeoer) [reply pkg/converter/types.go:119]: Updated the comment, PTAL.

Comment (changweige) [reply pkg/converter/types.go:119]: Thanks for the update, looks reasonable.

**Review by changweige — APPROVED** [review]
LGTM

**Pull request merged**